server.go

```diff
@@ -220,6 +220,11 @@ func (s *Server) Close() error {
 		return ErrServerClosed
 	}
 
+	// Stop goroutines.
+	if s.done != nil {
+		close(s.done)
+	}
+
```

Comment:
That's why it's blowing up? I guess I need wait groups then?

On Tuesday, February 10, 2015, dgnorton <notifications@github.com> wrote:

> In server.go
> <https://github.com/influxdb/influxdb/pull/1558#discussion_r24470055>:
>
> > @@ -220,6 +220,11 @@ func (s *Server) Close() error {
> >  		return ErrServerClosed
> >  	}
> >
> > +	// Stop goroutines.
> > +	if s.done != nil {
> > +		close(s.done)
> > +	}
> > +
>
> Need to wait on the goroutines to stop after this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/1558/files#r24470055>.
>